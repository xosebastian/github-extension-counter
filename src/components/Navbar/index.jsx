import * as React from 'react';
import { useState } from 'react';
import Grid from '@mui/material/Grid';
import TextField from '@mui/material/TextField';
import Button from '@mui/material/Button';
import Box from '@mui/material/Box';
import PropTypes from 'prop-types';
import { getUrl } from '../../utils';

export default function Navbar({ onClick }) {
  const [owner, setOwner] = useState('');
  const [repository, setRepository] = useState('');

  return (
    <Box sx={{ flexGrow: 1, mt: 2 }}>
      <Grid container spacing={1}>
        <Grid item xs={5}>
          <TextField
            fullWidth
            inputProps={{ 'data-testid': 'owner' }}
            size="small"
            autoComplete="off"
            label="Owner"
            variant="outlined"
            onChange={(e) => setOwner(e.target.value)}
          />
        </Grid>
        <Grid item xs={5}>
          <TextField
            size="small"
            autoComplete="off"
            fullWidth
            inputProps={{ 'data-testid': 'repository' }}
            label="Respository"
            variant="outlined"
            value={repository}
            onChange={(e) => setRepository(e.target.value)}
          />
        </Grid>
        <Grid item xs={2}>
          <Button
            fullWidth
            size="small"
            data-testid="button"
            onClick={() => onClick(getUrl(owner, repository))}
            variant="contained"
            disabled={!owner || !repository}
            style={{ lineHeight: 2.5 }}
          >
            Search
          </Button>
        </Grid>
      </Grid>
    </Box>
  );
}

Navbar.propTypes = {
  onClick: PropTypes.func.isRequired,
};
