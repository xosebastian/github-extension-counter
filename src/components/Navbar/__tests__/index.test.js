/**
 * @jest-environment jsdom
 */

import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Navbar from "../index";

const mockOnClick = jest.fn();

const getRenderedComponent = () => render(<Navbar onClick={mockOnClick} />);

describe("<Navbar/>", () => {
  it("Must render the component", () => {
    const { baseElement } = getRenderedComponent();
    expect(baseElement).toMatchSnapshot();
  });
  it("Expect to not log errors in console", () => {
    const spy = jest.spyOn(global.console, "error");
    getRenderedComponent();
    expect(spy).not.toHaveBeenCalled();
  });

  it("Should write the repository name", () => {
    getRenderedComponent();
    const repository = screen.getByTestId("repository");
    fireEvent.change(repository, { target: { value: "xosebastian" } });
    expect(repository.value).toBe("xosebastian");
  });
  it("Should write the owner name", () => {
    getRenderedComponent();
    const owner = screen.getByTestId("owner");
    fireEvent.change(owner, { target: { value: "lineaurbana" } });
    expect(owner.value).toBe("lineaurbana");
  });

  it("Must fire Click event for button", () => {
    getRenderedComponent();
    const repository = screen.getByTestId("repository");
    fireEvent.change(repository, { target: { value: "xosebastian" } });
    const owner = screen.getByTestId("owner");
    fireEvent.change(owner, { target: { value: "lineaurbana" } });
    const button = screen.getByTestId("button");
    fireEvent.click(button);
    expect(mockOnClick).toHaveBeenCalled();
  });
});
