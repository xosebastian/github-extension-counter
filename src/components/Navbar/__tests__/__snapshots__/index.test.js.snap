// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar/> Must render the component 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-be40y1"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r0:"
              id=":r0:-label"
            >
              Owner
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                data-testid="owner"
                id=":r0:"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Owner
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r1:"
              id=":r1:-label"
            >
              Respository
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                data-testid="repository"
                id=":r1:"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Respository
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-fullWidth MuiButtonBase-root Mui-disabled  css-1evprfk-MuiButtonBase-root-MuiButton-root"
            data-testid="button"
            disabled=""
            style="line-height: 2.5;"
            tabindex="-1"
            type="button"
          >
            Search
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
